src/containers/Main/Vote.js

```diff
@@ -89,21 +89,42 @@ function Vote({ history, getProposals }) {
 
   const updateBalance = async () => {
     if (account) {
-      const [currentVotes, balanceTemp, userInfo] = await Promise.all([
+      // find the pid of xvs vault, which users get voting powers from
+      const length = await xvsVaultProxyContract.methods
+        .poolLength(CONTRACT_XVS_TOKEN_ADDRESS)
+        .call();
+
+      const [currentVotes, balanceTemp, ...xvsPoolInfos] = await Promise.all([
         // voting power is calculated from user's amount of XVS staked in the XVS vault
         xvsVaultProxyContract.methods.getCurrentVotes(account).call(),
         xvsTokenContract.methods.balanceOf(account).call(),
-        xvsVaultProxyContract.methods
-          .getUserInfo(CONTRACT_XVS_TOKEN_ADDRESS, 0, account)
-          .call()
+        // query all xvs pool infos
+        ...Array.from({ length }).map((_, index) => {
+          return xvsVaultProxyContract.methods
+            .poolInfos(CONTRACT_XVS_TOKEN_ADDRESS, index)
+            .call();
+        })
       ]);
+
+      // find xvs vault pid
+      const xvsVaultIndex = xvsPoolInfos.findIndex(
+        info => info.token === CONTRACT_XVS_TOKEN_ADDRESS
```

Comment:
Someone may configure the address in an unchecked format, maybe sth like

```suggestion
        info => info.token.toLowerCase() === CONTRACT_XVS_TOKEN_ADDRESS.toLowerCase()
```
?